### ISSUE TITLE
Upgrade packages to WindowsAppSDK 1.4

### ISSUE THREAD

**Issue by LeonSpors** [CommunityToolkit/Labs-Windows#501]
### Describe the bug

The `CommunityToolkit.Labs.WinUI.*` packages are built on WindowsAppSDK 1.3. If you attempt to utilize WindowsAppSDK 1.4 in conjunction with any of the labs packages, you'll encounter numerous ambiguous reference errors.

### Steps to reproduce

```text
`dotnet add package Microsoft.WindowsAppSDK --version 1.4.231008000`
`dotnet add package CommunityToolkit.Labs.WinUI.Controls.DataTable --version 0.1.230830`
```

### Expected behavior

It should be compatible with the most recent WinAppSDK version.

### Code Platform

- [ ] UWP
- [X] WinAppSDK / WinUI 3
- [ ] Web Assembly (WASM)
- [ ] Android
- [ ] iOS
- [ ] MacOS
- [ ] Linux / GTK

### Windows Build Number

- [ ] Windows 10 1809 (Build 17763)
- [ ] Windows 10 1903 (Build 18362)
- [ ] Windows 10 1909 (Build 18363)
- [ ] Windows 10 2004 (Build 19041)
- [ ] Windows 10 20H2 (Build 19042)
- [ ] Windows 10 21H1 (Build 19043)
- [X] Windows 11 21H2 (Build 22000)
- [ ] Other (specify)

### App minimum and target SDK version

- [ ] Windows 10, version 1809 (Build 17763)
- [ ] Windows 10, version 1903 (Build 18362)
- [ ] Windows 10, version 1909 (Build 18363)
- [X] Windows 10, version 2004 (Build 19041)
- [ ] Other (specify)

### Visual Studio Version

2022

### Device form factor

Desktop

### Help us help you

Yes, I'd like to be assigned to work on this item.

**Comment by michael-hawker** [comment]
@LeonSpors once #485 is merged (hopefully soon), this should be easy to do.

Had mentioned some possibilities, like https://github.com/microsoft/microsoft-ui-xaml/issues/8810 in [Discord as well](https://discord.com/channels/372137812037730304/669640275500466197/1161669106911694949) [assuming that was you as well].

Are you trying to use .NET 7 as well in your project?

**Comment by LeonSpors** [comment]
Firstly, I want to apologize for the double message on both GitHub and Discord. We can verify that our project is utilizing .NET 7 along with Controls such as ItemsView.

I'm pleased to report that we were able to identify the issue, and it turned out to be a cascading problem. The main issue was related to the inclusion of the CommunityToolkit.WinUI.Converters package, which wasn't explicitly added as a NuGet package but was included transitively. This caused some problems, particularly because the CommunityToolkit.Labs.WinUI.Controls.DataTable package automatically brought in CommunityToolkit.Uno. However, we cannot definitively determine whether the source of the issue was a lack of package configuration or potentially triggered by Visual Studio. We know that we are using an experimental version where complete configuration may not be feasible.

The outcome of this was a multitude of ambiguities, causing the issues we encountered. However, since we've explicitly included the correct package, everything is now working smoothly with WindowsAppSDK 1.4 and .NET 7.

If you have any further questions or need additional information about this, please feel free to ask. Thanks for collaboration in resolving this issue.

**Comment by michael-hawker** [comment]
Thanks for the info @LeonSpors, that's helpful!

Currently, Labs is still built against earlier packages vs. the ones we shipped for 8.0 out of the `Windows` repo. Hopefully things can be a bit smoother using Labs components once we get this repo back up to speed as well.

@Arlodotexe we probably want another PR/issue to track updating all the dependencies of Labs components to use the shipped 8.0 packages as well, as I think we're still pointing to the pre-release ones?

**Comment by michael-hawker** [comment]
If there's a new WinUI 2 update, we should bump that as well.
